### ISSUE TITLE
When will IPv6 support be added?

### ISSUE THREAD

**Issue by dfarr24** [wireapp/wire-ios#2389]
So i submitted a ticket last week and was helped by Wire support and they determined that my problem was due to my cellular network provider using IPv6. (Background: my issue is that audio and video calling do not work when connected to the cellular network).

Apparently Wire does not support IPv6 yet, although it is being worked on. 

Any idea of when it will be added?

**Comment by mikeger** [comment]
Hey @dfarr24 ! Thanks for the question, I believe @jspittka can give more details on this topic.

**Comment by jspittka** [comment]
There are many different layers of IPv6 networks and how they interact with the remaining IPv4 world. About 2 years ago, Apple required app developers to integrate something that is called NAT64. This allows Apple devices that are connected to an IPv6 network to interface to the IPv4 world via a NAT64 gateway which is provided by the network service provider. Essentially this is supposed to ease the transition to IPv6 networks over time as obviously not everybody can make that switch at the same time. For example, most Android devices will stay on IPv4 for now.

Since then we have not seen any network providers switching their networks to an IPv6 infrastructure. Only recently we noticed that T-Mobile in the US switched to IPv6 for Apple devices and apparently now also your Canadian network service provider. In the early days of the switch for T-Mobile the NAT64 gateway did not work properly and users had similar issues as the one you reported. Fortunately, within a few days/weeks after those reports Wire calls started to work again and we suspect that some issue with the T-Mobile NAT64 gateway was fixed. 

It's difficult to say what exactly is causing the issue for you. What we can see in the log file you provided is that the NAT64 gateway is not working properly for you though. A possible solution is to not rely on the infrastructure of the network service provider but to provide a gateway between different IP worlds ourselves. This is something that we are actively working on but it will take some time and changes to our architecture. Especially proper testing will be essential.

So, in summary, Wire supports the most common IPv6 network scenario(s) but is dependent on that network service providers provide the proper infrastructure. We are working to make ourselves independent of that requirement but it will require some time. I expect this to happen within the next few month(s).

**Comment by dfarr24** [comment]
Thank you for the detailed explanation. While I am disappointed in this not working, I do very much enjoy using Wire and will continue to do so for texting purposes. Hopefully this issue will be rectified in the next few months as you say.
